hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -1325,6 +1325,7 @@ private GetBlockLocationsResult getBlockLocationsInt(final String srcArg, final
         }
       }
     }
+    inodeFile.logProvenanceEvent(FileProvenanceEntry.Operation.getBlockLocations());
```

Comment:
Do we support this for now? the operation is run with a READ lock so it won't work

Reply:
the READ operation does not rely on the inode logical time. all reads between two different logical times of an inode are treated as reading the same data, so it doesn't matter if they commute.
In epipe the timestamp and tie breaker field will provide total order (but logically for the provenance part i do not care of the exact order of reads with the same inode logical time)